pennylane/devices/default_gaussian.py

```diff
@@ -728,6 +730,13 @@ def apply(self, operation, wires, par):
             self._state = self._operation_map[operation](*par, hbar=self.hbar)
             return  # we are done here
 
+        if operation == "Snapshot":
+            self.snapshots[len(self.snapshots)] = {
```

Comment:
You cannot tag operation for this gaussian simulator?

Reply:
Yeah so unfortunately the way the `execute` function is written in the general device class it does not allow access to the operation object itself in the `apply` function, only to the name, wires, and differentiable parameters:
https://github.com/PennyLaneAI/pennylane/blob/d4931e715731fe84ff946ce1c4147a4db80c302d/pennylane/_device.py#L419-L420
The qubit device class has overwritten the `execute` function and does pass the operation object to `apply`, allowing access to other properties in those simulators.
https://github.com/PennyLaneAI/pennylane/blob/d4931e715731fe84ff946ce1c4147a4db80c302d/pennylane/_qubit_device.py#L200-L201
To support the tag in the Gaussian device we would need to do something similar or otherwise come up with a workaround to access the tag :/

Reply:
Ok @dime10 thank you for the explanation! I think it is ok to not implement it for now and maybe open an issue. I think it would be a great small project for external contributors.

Reply:
The inconsistency between the two execute methods is long overdue to be fixed 😬

Reply:
I'll open an issue for it :)

Reply:
Oh this is going to be fixed in the new prototype device API, which will replace the old device classes :)